### PR TITLE
VB-3514 Flyway script database changes

### PR DESCRIPTION
**Pull request by AledWEvans**
Flyway script database changes, removal of code that is no longer needed also api changes

### CODE REVIEW

**Review by dhirajshettymoj — APPROVED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/dto/VisitorSupportDto.kt [hunk 1]
```diff
@@ -2,25 +2,23 @@ package uk.gov.justice.digital.hmpps.visitscheduler.dto
...
 
 @Schema(description = "Visitor support")
 data class VisitorSupportDto(
-  @Schema(description = "Support type", example = "OTHER", required = true)
-  @field:NotBlank
-  val type: String,
   @Schema(description = "Support text description", example = "visually impaired assistance", required = false)
-  val text: String? = null,
+  @Size(min = 3, max = 512)
+  @NotBlank
```
Comment (dhirajshettymoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/dto/VisitorSupportDto.kt:13]: does there need to be any validation (probably a different story)

**Review by dhirajshettymoj — APPROVED** [review]
LGTM - comment added around validation but not necessarily part of this story

**Comment by dhirajshettymoj** [comment]
We can also remove caching now and the @EnableCaching annotation and associated dependencies.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/model/entity/VisitSupport.kt [hunk 2]
```diff
@@ -26,15 +26,12 @@ class VisitSupport(
   val id: Long = 0,
 
   @Column(name = "VISIT_ID", nullable = false)
-  var visitId: Long,
+  val visitId: Long,
 
-  @Column(name = "TYPE", nullable = false)
-  var type: String,
+  @Column(name = "DESCRIPTION", nullable = false)
+  var description: String? = null,
```
Comment (dhirajshettymoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/model/entity/VisitSupport.kt:32]: nullable is false.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/dto/VisitorSupportDto.kt [hunk 3]
```diff
@@ -2,25 +2,23 @@ package uk.gov.justice.digital.hmpps.visitscheduler.dto
...
+  @Schema(description = "Support text description", example = "visually impaired assistance", required = true)
+  @Size(min = 3, max = 512)
+  @NotBlank
+  val description: String,
 ) {
 
   constructor(entity: VisitSupport) : this(
-    type = entity.type,
-    text = entity.text,
+    description = entity.description!!,
```
Comment (dhirajshettymoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/dto/VisitorSupportDto.kt:18]: might not be needed - based on below comment.

**Review by dhirajshettymoj — COMMENTED** [review]
minor comments added

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/model/entity/VisitSupport.kt (hunk 2 above)
Comment (AledWEvans) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/model/entity/VisitSupport.kt:32]: Yes this is correct if there is a visit support it must have a description

**Review by dhirajshettymoj — APPROVED** [review]
LGTM

**Pull request merged**